xarray/core/dataset.py

```diff
@@ -1125,27 +1200,25 @@ def isel(self, drop=False, **indexers):
         See Also
         --------
         Dataset.sel
-        Dataset.sel_points
-        Dataset.isel_points
         DataArray.isel
         """
-        invalid = [k for k in indexers if k not in self.dims]
-        if invalid:
-            raise ValueError("dimensions %r do not exist" % invalid)
+        indexers_list = self._validate_indexers(indexers)
 
-        # all indexers should be int, slice or np.ndarrays
-        indexers = [(k, (np.asarray(v)
-                         if not isinstance(v, integer_types + (slice,))
-                         else v))
-                    for k, v in iteritems(indexers)]
+        coord_vars = self._get_indexers_coordinates(indexers)
+        coord_names = set(self._coord_names) | set(coord_vars)
```

Comment:
can we consolidate this with setting `coord_names` below, so we only set this once?

Reply:
Done